api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -70,9 +70,9 @@
     // conceptSetId and workspaceId
     ConceptSet conceptSet =
         conceptSetService.getConceptSet(dbWorkspace.getWorkspaceId(), conceptSetId);
-    conceptSetService.delete(conceptSet.getId());
     userRecentResourceService.deleteConceptSetEntry(
```

Comment:
Would it be possible to push this down into the service instead?

That way any call to `conceptSetService.delete()` will also delete its RecentResource.

Be careful to only delete if it exists, however.

Reply:
Will it be ok if i can tackle this in a new PR, since the other calls like userRecentResourceService . userRecentResourceService will also move into the conceptsetservice from controller

Reply:
Yeah I like that plan